dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/repository/impl/TaskInstanceContextDaoImpl.java

```diff
@@ -91,7 +90,7 @@ public int upsertTaskInstanceContext(TaskInstanceContext taskInstanceContext) {
             taskInstanceContext.setTaskInstanceContext(deduplicatedDependentResultTaskInstanceContextList);
             return mybatisMapper.updateTaskInstanceContextByTaskInstanceIdAndContextType(
                     taskInstanceContext.getTaskInstanceId(),
-                    taskInstanceContext.getContextType(), JSONUtils.toJsonString(taskInstanceContext.getContext()));
+                    taskInstanceContext.getContextType(), taskInstanceContext.getContext());
```

Comment:
It seems we can directly update by dbTaskInstanceContext's id?

Reply:
Yes. I've change it.